### PR TITLE
Logging for X509Cert test failure

### PR DESCRIPTION
**Pull request by danmosemsft**
Logging for https://github.com/dotnet/corefx/issues/6302

@bartonjs @steveharter

### CODE REVIEW

File: src/System.Security.Cryptography.X509Certificates/tests/PublicKeyTests.cs [hunk 1]
```diff
@@ -208,8 +210,28 @@ private static void VerifyKey_RSA(RSA rsa)
...
+            {
+                rsaParameters = rsa.ExportParameters(false);
+
+                if (!expectedModulus.SequenceEqual(rsaParameters.Modulus) ||
+                    !expectedExponent.SequenceEqual(rsaParameters.Exponent))
+                {
+                    Assert.False(true, "Second call to ExportParameters produced valid data, but the first did not");
+                }
+
+                rsa = cert.GetRSAPublicKey();
```
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/tests/PublicKeyTests.cs]: Presumably this should be in an `if (cert != null)`?

File: src/System.Security.Cryptography.X509Certificates/tests/PublicKeyTests.cs [hunk 2]
```diff
@@ -208,8 +210,28 @@ private static void VerifyKey_RSA(RSA rsa)
...
+                rsaParameters = rsa.ExportParameters(false);
+
+                if (!expectedModulus.SequenceEqual(rsaParameters.Modulus) ||
+                    !expectedExponent.SequenceEqual(rsaParameters.Exponent))
+                {
+                    Assert.False(true, "New key handle ExportParameters was successful, original key handle was not");
+                }    
+
+                Assert.False(true, "Key's busted");
```
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/tests/PublicKeyTests.cs]: Might need a better message here :).  Maybe even the original Assert.Equal (though maybe not, so we can tell we have the new logging?)

**Comment by bartonjs** [comment]
(Also: Thanks, Dan)

**Comment by danmosemsft** [comment]
OK, fixed the logging.
Seems the "right" way to emit messages is [ITestOutputHelper](https://xunit.github.io/docs/capturing-output.html) but that doesn't seem to work - at least it doesn't go to the console.
And Assert.Equal overloads don't accept a message either!

**Comment by danmosemsft** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release Build and Test (error below)

@mmitche have you seen this? `EXEC : error : Failed to download package from 'https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.build.framework/0.1.0-preview-00022/microsoft.build.framework.0.1.0-preview-00022.nupkg'. [/mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_release_prtest/src/Tools/CoreFx.Tools/CoreFx.Tools.csproj]`

**Comment by bartonjs** [comment]
@danmosemsft The `Assert.False(true, msg)` calls would have caused the test to fail with the message `msg`, so that one didn't have to hunt in the logs. (Well, and "Expected: False, Actual: True")

**Comment by danmosemsft** [comment]
Right, but then you wouldn't get the dump of the values that the Assert..Equals gives you.

**Comment by mmitche** [comment]
@danmosemsft Looks like perhaps a myget failure?  Is there retry functionality where that call is being made (probably buildtools)

/fyi @dotnet/dnceng

**Comment by dagood** [comment]
Here's a link to the failed test for more info: https://ci.dot.net/job/dotnet_corefx/job/master/job/centos7.1_release_prtest/2292/

We don't have any retry logic for the entire restore command, only the built-in retries that `restore` does: https://github.com/dotnet/buildtools/blob/e19e08d2ec82c8d95a3a4c68c573d08113680ea2/src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets#L35

This is a network timeout failure, which could be a MyGet failure.

**Pull request merged**